demo-client/src/main/java/com/illia/client/controller/DemoClientController.java

```diff
@@ -39,4 +43,15 @@ public ResponseEntity<String> downloadFile(@RequestParam(name = "fileName") Stri
     public ResponseEntity<String> deleteFile(@RequestParam(name = "fileName") String fileName) {
         return fileTransferService.deleteFile(fileName);
     }
+
+    @GetMapping("/query")
+    public ResponseEntity<Object> performOperation(@RequestParam Map<String, String> params) {
+        var response = queryProcessingService.performOperation(params);
```

Comment:
It is better to compose a response at the controller level, not the service one
```suggestion
        return ResponseEntity.ok(queryProcessingService.performOperation(params));
```